crowbar_framework/app/models/service_object.rb

```diff
@@ -978,7 +997,17 @@ def apply_role(role, inst, in_queue, bootstrap = false)
     else
       # Attempt to queue the proposal.  If delay is empty, then run it.
       deps = proposal_dependencies(role)
-      delay, pre_cached_nodes = queue_proposal(inst, element_order, new_elements, deps)
+      if skip_unready_nodes_enabled
+        elements_without_unready, pre_cached_nodes = skip_unready_nodes(
```

Comment:
We're still missing something here. We need to have a line that looks like `new_elements = elements_without_unready` at some point, but after `new_deployment["elements_expanded"] = new_elements` and also we need to make sure to expand all clusters in `elements_without_unready` when we do this.

Without this, we will still try to create the chef lock and run chef-client on unready nodes, which will fail if one of the unready nodes is just down.